### PR TITLE
Pause pool using `unpaused` modifier

### LINKED ISSUE #207: Add per market functionality to pause()

### LINKED ISSUE #324: Add per-protocol (or per market might be better?) pause functionality to lender
If we do it per market, we should implement a batch pause with an array so that you can still pause all of a protocol at once.

### PR DESCRIPTION
**Pull request by sourabhmarathe**
See #331.

This PR implements a flag that pauses usage of the pools.

### CODE REVIEW

File: build/marketplace/MarketPlace.sol [hunk 1]
```diff
@@ -34,6 +34,8 @@ contract MarketPlace {
     address public admin;
     /// @notice address of the deployed redeemer contract
     address public immutable redeemer;
+    /// @notice flag that determines if pools may be used
+    bool public paused;
```
Comment (robrobbins) [inline build/marketplace/MarketPlace.sol]: this, obvs, is all markets

do we want this blanket pause or an individual way to pause a specific market?

Comment (sourabhmarathe) [reply build/marketplace/MarketPlace.sol]: I'm not sure. I'll ask product.

File: build/marketplace/MarketPlace.sol [hunk 2]
```diff
@@ -88,6 +88,15 @@ contract MarketPlace {
         return true;
     }
 
+    /// @notice allows the admin to pause a protocol's pools
+    /// @param p protocol's enum value
```
Comment (robrobbins) [inline build/marketplace/MarketPlace.sol]: PRINCIPAL not PROTOCOL

File: build/marketplace/MarketPlace.sol [hunk 3]
```diff
@@ -177,4 +186,9 @@ contract MarketPlace {
         require(msg.sender == a, 'sender must be authorized');
         _;
     }
+
+    modifier unpaused(uint8 p) {
+        require(!paused[p], 'markets are paused');
```
Comment (robrobbins) [inline build/marketplace/MarketPlace.sol]: i think the err msg should be `principal is paused` or something with the fact that its Principal based?

File: build/marketplace/MarketPlace.sol [hunk 4]
```diff
@@ -34,8 +34,8 @@ contract MarketPlace {
     address public admin;
     /// @notice address of the deployed redeemer contract
     address public immutable redeemer;
-    /// @notice flag that determines if pools may be used
-    bool public paused;
+    /// @notice flag that determines if a protocol's pool is available
+    bool[9] public paused = [false, false, false, false, false, false, false, false, false];
```
Comment (robrobbins) [inline build/marketplace/MarketPlace.sol:38]: using the principals enum -- i liek it.

**Review by robrobbins — CHANGES_REQUESTED** [review]

**Pull request merged**